kratos/sources/model_part_io.cpp

```diff
@@ -179,18 +179,37 @@ namespace Kratos
         std::string aux_string;
         const std::string string_to_remove = "This properties contains 0 tables";
 
-        for(PropertiesContainerType::const_iterator i_properties = rThisProperties.begin() ; i_properties != rThisProperties.end() ; i_properties++)
-        {
+        for (const auto i_properties = rThisProperties.begin() ; i_properties != rThisProperties.end() ; ++i_properties) {
             std::ostringstream aux_ostream;
             (*mpStream) << "Begin Properties " << i_properties->Id() << std::endl;
             i_properties->PrintData(aux_ostream);
 
             aux_string = aux_ostream.str();
-            std::string::size_type i = aux_string.find(string_to_remove);
 
-            if (i != std::string::npos)
-            {
-                aux_string.erase(i, string_to_remove.length());
+            // We remove the line of Constitutive Laws and we add it manually
+            if (i_properties->Has(CONSTITUTIVE_LAW)) {
+                std::string::size_type it_constitutive_law_begin = aux_string.find("CONSTITUTIVE_LAW");
+                std::string::size_type it_constitutive_law_end = aux_string.find("\n", it_constitutive_law_begin);
+
+                if (it_constitutive_law_begin != std::string::npos) {
+                    aux_string.erase(it_constitutive_law_begin, it_constitutive_law_end);
```

Comment:
why do you need to erase it? i see it was done before, i just don't see why ... (and i don't love it)

Reply:
Because the line it returns the Info includes CONTITUTIVE_LAW and the pointer, I want to remove that line and add the proper one

Reply:
I can add a description if you want

Reply:
Done, more description